src/System.Memory/src/System/SpanHelpers.byte.cs

```diff
@@ -122,15 +122,16 @@ public static unsafe int IndexOf(ref byte searchSpace, byte value, int length)
                 {
                     goto NotFound;
                 }
-                nLength = (IntPtr)(uint)(length - Vector<byte>.Count);
+                nLength = (IntPtr)(uint)(length - (uint)index);
```

Comment:
Could be:
 `nLength = (IntPtr)(uint)((length - (uint)index) & ~(Vector<byte>.Count - 1));`

Then you don't need the second calc in the loop? And can compare against index?